### ISSUE TITLE
Make machine_id result consistant across Meterpreters

### ISSUE THREAD

**Issue by OJ** [rapid7/metasploit-framework#5441]
At the moment the PHP Meterpreter returns a different value for `machine_id` as per [this comment](https://github.com/rapid7/metasploit-framework/pull/5367#discussion_r30500235). The `machine_id` code on posix should do one of the following:

* Sort the list of entries in `/dev/disk/by-id/` before selecting the first one.
* Have some other way of making sure that the chosen entry from `/dev/disk/by-id/` is consistent.
* Use a completely different way of getting something unique off the machine that works consistently in each Meterpreter that runs on POSIX.

I'm open to suggestions!

**Comment by hmoore-r7** [comment]
@OJ @bcook-r7 @timwr After our latest sync on this, I think we agreed to allow for different machine_id implementations by architecture, given the challenge of consistent IDs across things like Java, PHP, and Android.

**Comment by OJ** [comment]
Yes, we definitely do, the problem we have here is that `machine_id` isn't the same on PHP Meterpreter as it is on, say, Python Meterpreter on the _same_ machine.

**Comment by bcook-r7** [comment]
I've been thinking about switching to the OS-generated UUID of the root partition as the eventual source of truth for machine UUID for Linux and others. This should be derivable on all POSIX operating systems these days, and given that there is only one, sorting order should not matter. There's just the simple matter of coming up with the right way to query it portably :) This probably assumes too much about the underlying Linux distro, for instance:

blkid -s UUID -o value `mount | awk '/ \/ / {print $1}'`

**Comment by MSadek-r7** [comment]
Hey @OJ , I've been taking a look at this ticket but haven't been able to recreate the bug. Could you provide some more info on the circumstances which would lead to a mismatching machine_id result?

**Comment by void-in** [comment]
@MSadek-r7 Trying to chip in for @OJ here since he is working on a lot of tickets. So to test, generate two different meterpreters (e.g. python and PHP), run it on the same machine and look at the machine ID passed to the handler. You can view the machine ID using the sessions -l -v switch.

Since the machine is same, ideally the machine ID should be same across different meterpreter sessions as well no matter if it is python, PHP or native.

Also, querying such privileged information as HDD UUID on a PHP meterpreter on Windows is not possible. The ticket is coming up with a solution which can be used to generate consistent machine IDs across all the meterpreter architectures.

**Comment by OJ** [comment]
Thanks @void-in! This description is spot on.

@MSadek-r7 I've not seen it on all platforms, though I had it break on my Fedora 21 x64 install. I'll do another test for you later though.

**Comment by OJ** [comment]
With the change above, things don't go well:
```
payload => windows/meterpreter_reverse_http
LHOST => 10.1.10.40
LPORT => 9000
[*] Started HTTP reverse handler on http://0.0.0.0:9000/
[*] Starting the payload handler...
[*] 10.1.10.35:49526 (UUID: a8993b5f37453093/x86=1/windows=1/2015-07-01T21:56:12Z) Redirecting stageless connection ...
[-] Exception handling request: uninitialized constant Msf::Handler::ReverseHttp::PACKET_TYPE_RESPONSE
[*] 10.1.10.35:49527 (UUID: a8993b5f37453093/x86=1/windows=1/2015-07-01T21:56:12Z) Redirecting stageless connection ...
[-] Exception handling request: uninitialized constant Msf::Handler::ReverseHttp::PACKET_TYPE_RESPONSE
```
I've added the full namespace to `PACKET_TYPE_RESPONSE` and now it's good:
```
payload => windows/meterpreter_reverse_http
LHOST => 10.1.10.40
LPORT => 9000
[*] Started HTTP reverse handler on http://0.0.0.0:9000/
[*] Starting the payload handler...
[*] 10.1.10.35:49570 (UUID: a8993b5f37453093/x86=1/windows=1/2015-07-01T21:56:12Z) Redirecting stageless connection ...
[*] 10.1.10.35:49571 (UUID: a8993b5f37453093/x86=1/windows=1/2015-07-01T21:56:12Z) Attaching orphaned/stageless session ...
[*] Meterpreter session 1 opened (10.1.10.40:9000 -> 10.1.10.35:49571) at 2015-07-02 08:02:04 +1000

meterpreter > sysinfo
Computer        : WIN-S45GUQ5KGVK
OS              : Windows 7 (Build 7601, Service Pack 1).
Architecture    : x64 (Current Process is WOW64)
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x86/win32
```
Landing.

**Comment by fsacer** [comment]
updates? @OJ
